pennylane/beta/tapes/qnode.py

```diff
@@ -383,11 +383,38 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+
+        if self.interface == "autograd":
+            # HOTFIX: to maintain compatibility with core, here we treat
+            # all inputs that do not explicitly specify `requires_grad=False`
+            # as trainable. This should be removed at some point, forcing users
+            # to specify `requires_grad=True` for trainable parameters.
+            args = [
+                anp.array(a, requires_grad=True) if not hasattr(a, "requires_grad") else a
+                for a in args
+            ]
+
         # construct the tape
         self.construct(args, kwargs)
 
         # execute the tape
-        return self.qtape.execute(device=self.device)
+        res = self.qtape.execute(device=self.device)
+
+        if list(res.shape) == [1]:
+            # HOTFIX: to maintain compatibility with core, we squeeze
+            # any output that has shape (1,).
+
+            # Get the namespace associated with the return type
+            res_type_namespace = res.__class__.__module__.split(".")[0]
+
+            if res_type_namespace in ("pennylane", "autograd"):
+                # For PennyLane and autograd we must branch, since
+                # 'squeeze' does not exist in the top-level of the namespace
+                return anp.squeeze(res)
+
+            return __import__(res_type_namespace).squeeze(res)

```

Comment:
Could this not be done in the interfaces, where they overwrite the execute function?

Reply:
I originally thought the same, and had tried to do this.

I added `._squeeze` attributes to the interfaces, and had `AutogradInterface._squeeze(res)` here.

Unfortunately, this doesn't work for 'passthru' devices though, since they don't have an 'interface', they are purely written in the ML framework already!

Reply:
I always forget about those. Sure!

Reply:
An alternative would be to give passthru devices a dummy interface that just does some post-processing...just a thought. So everything is on the same footing.

Reply:
Not sure if it is worth the extra overhead/maintenance... the main purpose of the interfaces is to provide custom gradient wrappers, which the backprop devices do not need